bitwarden_license/src/Commercial.Infrastructure.EntityFramework/SecretsManager/Repositories/SecretRepository.cs

```diff
@@ -252,4 +243,70 @@ public async Task<IEnumerable<Core.SecretsManager.Entities.Secret>> ImportAsync(
         }
         return secrets;
     }
+
+    public async Task<(bool Read, bool Write)> AccessToSecretAsync(Guid id, Guid userId, AccessClientType accessType)
+    {
+        using var scope = ServiceScopeFactory.CreateScope();
+        var dbContext = GetDatabaseContext(scope);
+
+        var accessPolicy = await dbContext.Secret
+            .Where(s => s.Id == id)
+            .Select(s => new
+            {
+                Read = s.Projects.Any(p =>
+                    p.UserAccessPolicies.Any(ap => ap.OrganizationUser.User.Id == userId && ap.Read) ||
+                    p.GroupAccessPolicies.Any(ap =>
+                        ap.Group.GroupUsers.Any(gu => gu.OrganizationUser.User.Id == userId && ap.Read))),
+                Write = s.Projects.Any(p =>
+                    p.UserAccessPolicies.Any(ap => ap.OrganizationUser.User.Id == userId && ap.Write) ||
+                    p.GroupAccessPolicies.Any(ap =>
+                        ap.Group.GroupUsers.Any(gu => gu.OrganizationUser.User.Id == userId && ap.Write))),
+            }).FirstOrDefaultAsync();
+
+        return (accessPolicy.Read, accessPolicy.Write);
+    }
+
+    private IQueryable<SecretPermissionDetails> SecretPermissionDetailsEnumerable(IQueryable<Secret> query, Guid userId, AccessClientType accessType)
+    {
+        var secrets = accessType switch
+        {
+            AccessClientType.NoAccessCheck => query.Select(s => new SecretPermissionDetails
+            {
+                Secret = Mapper.Map<Bit.Core.SecretsManager.Entities.Secret>(s),
+                Read = true,
+                Write = true,
+            }),
+            AccessClientType.User => query.Where(UserHasReadAccessToSecret(userId)).Select(SecretToPermissionsUser(userId)),
+            AccessClientType.ServiceAccount => query.Where(ServiceAccountHasReadAccessToSecret(userId)).Select(s =>
+                new SecretPermissionDetails
+                {
+                    Secret = Mapper.Map<Bit.Core.SecretsManager.Entities.Secret>(s),
+                    Read = true,
+                    Write = false,
+                }),
+            _ => throw new ArgumentOutOfRangeException(nameof(accessType), accessType, null),
+        };
+        return secrets;
+    }
+
+    private Expression<Func<Secret, SecretPermissionDetails>> SecretToPermissionsUser(Guid userId) =>
+        s => new SecretPermissionDetails
+        {
+            Secret = Mapper.Map<Bit.Core.SecretsManager.Entities.Secret>(s),
+            Read = true,
```

Comment:
In the context of where this private function is currently being used, hard coding `Read = true` makes sense. 

However, from a consumer of this private function perspective, I'm expecting it should calculate both. 

What do you think about modifying the `SecretToPermissionsUser(Guid userId)` to `SecretToPermissionsUser(Guid userId, bool read)` so it at least indicates to the consumer they will need to provide the Read value?